src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -139,9 +150,8 @@ private static Project CreateProject(ProjectId projectId, Solution solution)
             CancellationToken cancellationToken = default)
 #pragma warning restore IDE0060 // Remove unused parameter
         {
-            var projectState = _state.GetProjectState(assemblySymbol);
```

Comment:
this became cleaner as a static helper (assembly-to-projectid) and *then* calling the normal GetProject helper.